lib/module/doom.go

```diff
@@ -0,0 +1,55 @@
+package module
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/dvdmuckle/irkbot/lib/configure"
+	"github.com/dvdmuckle/irkbot/lib/message"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type doomStruct struct {
+	Type string `json:"type"`
+}
+
+const doomHost = "http://localhost:8000/api/player/actions"
+
+func HelpDoom() []string {
+	s := "doom <command> - play doom!"
+	return []string{s}
+}
+
+//Sanitize input
+func Doom(cfg *configure.Config, in *message.InboundMsg, actions *Actions) {
+	doomCommand := "enter a command, dipstick"
+	if len(in.MsgArgs[1:]) == 0 {
+		actions.Say(doomCommand)
+		return
+	}
+	//Ugly switch statement to sanitize input because RESTful Doom doesn't
+	doomCommand = strings.Join(in.MsgArgs[1:], " ")
+	switch doomCommand {
```

Comment:
I wouldn't do a switch statement here because you now have to define the list of doom commands in two places: in the switch statement itself, and in the error message you send back to the client. It would be really easy to update one and forget to update the other, not to mention that there's no guarantee that the spelling is consistent between the two.

I would store the list of valid commands in an array. That way, you can iterate through it to check the validity of the input, and then use that same array to return the list of valid commands.